jerry-core/ecma/builtin-objects/ecma-builtin-date.c

```diff
@@ -198,91 +197,132 @@ static ecma_value_t
 ecma_date_construct_helper (const ecma_value_t *args, /**< arguments passed to the Date constructor */
                             ecma_length_t args_len) /**< number of arguments */
 {
-  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
-  ecma_number_t prim_value = ecma_number_make_nan ();
+  /* 1. */
+  ecma_value_t year_value = ecma_op_to_number (args[0]);
```

Comment:
How about using `ecma_get_number`? It should require less `ecma_free_value` calls.